ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -614,6 +614,24 @@ private FunctionDescriptionList describeNonAggregateFunction(
     );
   }
 
+  private KsqlEntity executeRunScript(
+      final PreparedStatement<RunScript> statement,
+      final Map<String, Object> propertyOverrides
+  ) {
+    final String sql = (String) propertyOverrides
+        .get(KsqlConstants.LEGACY_RUN_SCRIPT_STATEMENTS_CONTENT);
+
+    if (sql == null) {
+      throw new KsqlStatementException(
+          "Request is missing script content", statement.getStatementText());
+    }
+
+    final KsqlEntityList entities = new KsqlEntityList();
+    ksqlEngine.parse(sql).forEach(stmt -> executeStatement(stmt, propertyOverrides, entities));
+
+    return entities.isEmpty() ? null : Iterables.getLast(entities);
```

Comment:
Why do we only add the last entity? Doesn't this mean that if a RUN SCRIPT contains multiple CREATE statements, in the response it'll look like only the last one succeeded?

Reply:
This is a hack for backwards compatibility. Previously, runscript would be posted as a single message to the command topic, and so users would only expect a single response. 

It's not ideal, and there will be edge cases, but this is all being stripped out from 6.0 anyway. It's just a temp work around.